src/main/xsl/preprocess/mappullImpl.xsl

```diff
@@ -499,7 +499,7 @@ Other modes can be found within the code, and may or may not prove useful for ov
 
           <!--finding type based on name of the target element in a particular topic in another file-->
           <xsl:when test="$topicpos='otherfile'">
-            <xsl:variable name="target" select="$doc//*[@id=$topicid]" as="element()?"/>
+            <xsl:variable name="target" select="$doc//*[contains(@class,' topic/topic ')][@id=$topicid][1]" as="element()?"/>
```

Comment:
I think that sounds reasonable - did not occur to me since I was just tweaking the existing logic. I think the result should be the same, and slightly more efficient.